net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/BasicWCClient.java

```diff
@@ -0,0 +1,282 @@
+			}
+			data = root.get("sunsetTimeUtc");
+			if ( data.isArray() ) {
+				current.setSunset(parseTimeValue(data.get(i)));
+
+			}
+			data = root.get("moonriseTimeUtc");
+			if ( data.isArray() ) {
+				current.setMoonrise(parseTimeValue(data.get(i)));
+			}
+			data = root.get("moonsetTimeUtc");
+			if ( data.isArray() ) {
+				current.setMoonset(parseTimeValue(data.get(i)));
+			}
+			data = root.get("validTimeUtc");
+			if ( data.isArray() ) {
+				current.setCreated(new Date(data.get(i).asLong() * 1000));
+			}
+			data = root.get("temperatureMax");
+			if ( data.isArray() ) {
+				current.setTemperatureMaximum(parseBigDecimal(data.get(i)));
+			}
+			data = root.get("temperatureMin");
+			if ( data.isArray() ) {
+				current.setTemperatureMinimum(new BigDecimal(data.get(i).asLong()));
+			}
+
+		}
+
+		log.debug("Finished IBM Weather retrieval");
+		return result;
+	}
+
+	@Override
+	public Collection<WCHourlyDatum> readHourlyForecast(String locationIdentifier, String apiKey,
+			String datumPeriod) {
+		if ( locationIdentifier == null ) {
+			return null;
+		}
+		final List<WCHourlyDatum> result = new ArrayList<WCHourlyDatum>();
+		final String url = getURL(this.hourlyForecastUrl, locationIdentifier, apiKey, datumPeriod);
+		JsonNode root;
+		try {
+			log.debug("opening IBM Weather API connection");
+			log.debug(url);
+			URLConnection conn = getURLConnection(url, HTTP_METHOD_GET);
+			root = getObjectMapper().readTree(getInputStreamFromURLConnection(conn));
+		} catch ( IOException e ) {
+			log.warn("Error reading WC URL [{}]: {}", url, e.getMessage());
+			return result;
+		}
+
+		for ( JsonNode n : root.get("dayOfWeek") ) {
+			result.add(new WCHourlyDatum());
+		}
+
+		for ( int i = 0; i < result.size(); i++ ) {
+
+			WCHourlyDatum current = result.get(i);
+			JsonNode data = root.get("validTimeUtc");
+			if ( data.isArray() ) {
+				current.setCreated(new Date(data.get(i).asLong() * 1000));
+			}
+			data = root.get("temperature");
+			if ( data.isArray() ) {
+				current.setTemperature(parseBigDecimal(data.get(i)));
+			}
+			data = root.get("visibility");
+			if ( data.isArray() ) {
+				current.setVisibility(parseInt(data.get(i)));
+			}
+			data = root.get("windDirection");
+			if ( data.isArray() ) {
+				current.setWindDirection(parseInt(data.get(i)));
+			}
+			data = root.get("windSpeed");
+			if ( data.isArray() ) {
+				current.setWindSpeed(parseBigDecimal(data.get(i)));
+			}
+			data = root.get("relativeHumidity");
+			if ( data.isArray() ) {
+				current.setHumidity(parseInt(data.get(i)));
+			}
+			data = root.get("cloudCover");
+			if ( data.isArray() ) {
+				current.setCloudCover(parseBigDecimal(data.get(i)));
+			}
+
+			//rainfall?
+			/*
+			 * data = root.get("precipChance"); if(data.isArray()) {
+			 * current.setRain(parseInt()); }
+			 */
+
+		}
+
+		log.debug("Finished IBM Hourly Weather retrieval");
+		return result;
+	}
+
+	public ObjectMapper getObjectMapper() {
```

Comment:
This getter, and associated setter, can be inherited from `JsonHttpClientSupport`.